### PR TITLE
Force program installations on all systems

### PR DESCRIPTION
**Pull request by Eprince-hub**
Make the installation commands for all systems use the `--force` flag to override any previous installations the student may have before starting the system setup

Using the `--yes --force` to prevent an already installed program/software from interfering with the new installations from the system setup on `Windows` and `macOS`

[Chocolatey Flags](https://docs.chocolatey.org/en-us/choco/commands/#default-options-and-switches)
[Homebrew Flags](https://docs.brew.sh/Manpage)

The `--reinstall` is used for the `apt` in Linux installation guide as it functions the same as `--force` used for other systems for `apt`. The `--force` flag doesn't exist for `apt`
[apt flags](https://manpages.ubuntu.com/manpages/noble/man8/apt-get.8.html)

### CODE REVIEW

File: macos.md [hunk 1]
```diff
@@ -36,7 +36,7 @@ Make sure that you're running the 2nd-newest macOS version or the newest version
    <img src="./macos-4.1-homebrew-next-steps.png"><br><br>
 6. Copy each line in the following text, paste it in the terminal and hit return.<br><br>
    ```bash
-   brew install flyctl git less node@20 python
+   brew install flyctl git less node@20 python -f
```
Comment (karlhorky) [inline macos.md:39]: 1. We use long flags (with `--`, full word 
2. It should be at the same position as the other flags that we use, not separate

File: windows.md [hunk 2]
```diff
@@ -31,7 +31,7 @@ With those compatibility things out of the way, you're ready to start the system
    <br>This will run Powershell as an administrator user<br><br>
 3. Copy the following text (be sure you select all of it, it's very long) and right-click in the blue middle part of the PowerShell window to paste the text. Hit enter.<br><br>
    ```bash
-   Set-ExecutionPolicy AllSigned -Force; [System.Net.ServicePointManager]::SecurityProtocol = [System.Net.ServicePointManager]::SecurityProtocol -bor 3072; iex ((New-Object System.Net.WebClient).DownloadString('https://chocolatey.org/install.ps1'))
+   Set-ExecutionPolicy AllSigned -f; [System.Net.ServicePointManager]::SecurityProtocol = [System.Net.ServicePointManager]::SecurityProtocol -bor 3072; iex ((New-Object System.Net.WebClient).DownloadString('https://chocolatey.org/install.ps1'))
```
Comment (karlhorky) [inline windows.md:34]: I guess this was a mistake?

File: windows.md [hunk 3]
```diff
@@ -40,19 +40,19 @@ With those compatibility things out of the way, you're ready to start the system
 5. Close PowerShell and open it again as administrator (like in step 2)<br><br>
 6. Copy the following text and right-click in the blue middle part of the PowerShell window to paste the text. Hit enter.<br><br>
    ```bash
-   choco install git nodejs-lts vscode hyper httpie flyctl -y
+   choco install git nodejs-lts vscode hyper httpie flyctl -y -f
```
Comment (karlhorky) [inline windows.md:43]: Let's switch to full flags `--yes --force`

https://docs.chocolatey.org/en-us/choco/commands/#default-options-and-switches

File: windows.md [hunk 4]
```diff
@@ -483,7 +483,7 @@ With those compatibility things out of the way, you're ready to start the system
     Copy each line in the following text, paste it in Hyper and hit return.
 
     ```bash
-    choco install androidstudio
+    choco install androidstudio -f
```
Comment (karlhorky) [inline windows.md:486]: Seems like some packages are missing the `--yes` flag, let's fix that in this PR too

Comment (Eprince-hub) [reply windows.md]: ac2b6b4326d4cf5f5d372d1b920fe2c0e902f263

File: windows.md (hunk 3 above)
Comment (Eprince-hub) [reply windows.md]: ac2b6b4326d4cf5f5d372d1b920fe2c0e902f263

File: windows.md (hunk 2 above)
Comment (Eprince-hub) [reply windows.md]: b6d2e80acae7eb03ba878745d18e9f10a4ae59c7

File: macos.md (hunk 1 above)
Comment (Eprince-hub) [reply macos.md:39]: ac2b6b4326d4cf5f5d372d1b920fe2c0e902f263

**Comment by karlhorky** [comment]
Hmm, just looking at [the Chocolatey best practices docs](https://docs.chocolatey.org/en-us/choco/commands/#:~:text=Never%20use%20the%20force%20option%20(%2D%2Dforce/%2Df)%20in%20scripts%20(or%20really%20otherwise%20as%20a%20default%20mode%20of%20use).%20Force%20is%20an%20override%20on%20Chocolatey%20behavior.%20If%20you%20are%20wondering%20why%20Chocolatey%20isn%E2%80%99t%20doing%20something%20like%20the%20documentation%20says%20it%20should%2C%20it%E2%80%99s%20likely%20because%20you%20are%20using%20force.%20Stop.):

> Never use the force option (--force/-f) in scripts (or really otherwise as a default mode of use). Force is an override on Chocolatey behavior. If you are wondering why Chocolatey isn’t doing something like the documentation says it should, it’s likely because you are using force. Stop.

So maybe we reconsider on this PR - alternatives would be to:

1. Detect what software already exists at the start as mentioned in #11 
2. Add a message that `--force` can be used to the end of the more problematic steps like those mentioned in [issue #97](https://github.com/upleveled/system-setup/issues/97) and [issue #98](https://github.com/upleveled/system-setup/issues/98), if it can be detected

**Comment by karlhorky** [comment]
Moved the long `--yes` flags to separate commit:

-

**Comment by karlhorky** [comment]
Let's close this for now, we can revisit [the alternatives above](https://github.com/upleveled/system-setup/pull/100#:~:text=So%20maybe%20we%20reconsider%20on%20this%20PR%20%2D%20alternatives%20would%20be%20to%3A) if/when we run into more installation problems with systems that already have software

**Pull request closed without merging**